pennylane/math/multi_dispatch.py

```diff
@@ -623,14 +624,16 @@ def where(condition, x=None, y=None):
     """
     if x is None and y is None:
         interface = _multi_dispatch([condition])
-        return np.where(condition, like=interface)
+        res = np.where(condition, like=interface)
+
+        if interface == "tensorflow":
+            return np.transpose(np.stack(res))
+
+        return res
 
     interface = _multi_dispatch([condition, x, y])
     res = np.where(condition, x, y, like=interface)
 
-    if interface == "tensorflow":
-        return np.transpose(np.stack(res))
-

```

Comment:
@josh146 I know autoray made a change related to this recently that broke some stuff. Right now all the tests are passing so it *should* be fine, but just double checking with you in case I still inadvertently broke something.

Reply:
Context: I had to rewrite `where` for TensorFlow in `single_dispatch.py` because the autoray implementation raised some error in AutoGraph mode.

Reply:
@eddddddy looks good to me if the tests pass :)